storage/src/tests/distributor/mergelimitertest.cpp

```diff
@@ -42,47 +42,53 @@ struct NodeFactory {
     operator const MergeLimiter::NodeArray&() const { return _nodes; }
 };
 
-#define ASSERT_LIMIT(maxNodes, nodes, result) \
-{ \
-    MergeLimiter limiter(maxNodes); \
-    auto nodesCopy = nodes; \
-    limiter.limitMergeToMaxNodes(nodesCopy); \
-    std::ostringstream actual; \
-    for (uint32_t i = 0; i < nodesCopy.size(); ++i) { \
-        if (i != 0) actual << ","; \
-        actual << nodesCopy[i]._nodeIndex; \
-        if (nodesCopy[i]._sourceOnly) actual << 's'; \
-    } \
-    ASSERT_EQ(result, actual.str()); \
 }
 
-}
+struct MergeLimiterTest : Test {
+
+    static std::string limit(uint32_t max_nodes, std::vector<MergeMetaData> nodes) {
```

Comment:
If code size/compile time reduction wanted, it might be necessary to implement it outside of the class definition.